### PR TITLE
feat: migrate store/auth to typescript 

### LINKED ISSUE #725: Convert to typescript - store/auth

### PR DESCRIPTION
**Pull request by zakariaelas**
fixes #725

### CODE REVIEW

File: packages/medusa-js/src/resources/auth.ts [hunk 1]
```diff
@@ -6,7 +8,7 @@ class AuthResource extends BaseResource {
    * @description Authenticates a customer using email and password combination
    * @returns AsyncResult<{ customer: Customer }>
    */
-  authenticate(payload: Types.AuthCreateSessionResource): Types.AsyncResult<{ customer: Types.Customer }> {
+  authenticate(payload: StorePostAuthReq): Types.AsyncResult<{ customer: Types.Customer }> {
```
Comment (kasperkristensen) [inline packages/medusa-js/src/resources/auth.ts:11]: Should have return type `AxiosPromise<StoreGetAuthRes>`

File: packages/medusa/src/api/routes/store/auth/index.ts [hunk 2]
```diff
@@ -17,3 +17,12 @@ export default (app) => {
 
   return app
 }
+
```
Comment (kasperkristensen) [inline packages/medusa/src/api/routes/store/auth/index.ts:20]: add `export class StoreGetAuthRes {
    customer: Customer
}`

**Review by kasperkristensen — CHANGES_REQUESTED** [review]

File: packages/medusa/src/api/routes/store/auth/index.ts (hunk 2 above)
Comment (zakariaelas) [reply packages/medusa/src/api/routes/store/auth/index.ts:21]: I went with `StoreAuthRes` as it made a tiny bit more sense + to stay a bit consistent with the naming conventions throughout the codebase. What do you think?

Comment (kasperkristensen) [reply packages/medusa/src/api/routes/store/auth/index.ts:21]: @zakariaelas I agree makes perfect sense :)

**Review by kasperkristensen — APPROVED** [review]
LGTM

**Pull request merged**